vignettes/ebola.Rmd

```diff
@@ -0,0 +1,615 @@
+In some of these models, we vary the parameters of the delay distribution by sex or by district.
+For the lognormal delay distribution, which we begin by focusing on, these parameters are the mean and standard deviation of the underlying normal distribution.
+That is, $\mu$ and $\sigma$ such that when $x \sim \mathcal{N}(\mu, \sigma)$ that $\exp(x)$ has a lognormal distribution.
+
+## Data preparation
+
+To prepare the data, we begin by transforming the date columns to `ptime` and `stime` columns for the times of the primary and secondary events respectively.
+Both of these columns are relative to the first date of symptom onset in the data.
+We also transform the `data.frame` to a `data.table`.
+
+```{r}
+sierra_leone_ebola_data <- sierra_leone_ebola_data |>
+  dplyr::mutate(
+    date_of_symptom_onset = lubridate::ymd(date_of_symptom_onset),
+    date_of_sample_tested = lubridate::ymd(date_of_sample_tested),
+    ptime = as.numeric(date_of_symptom_onset - min(date_of_symptom_onset)),
+    stime = as.numeric(date_of_sample_tested - min(date_of_symptom_onset))
+  ) |>
+  dplyr::select(case, ptime, stime, age, sex, district) |>
+  data.table::as.data.table()
+```
+
+The data now looks like:
+
+```{r}
+head(sierra_leone_ebola_data)
+```
+
+Next, we use `observe_process()` to add interval censoring columns giving the lower and upper bounds on the primary and secondary event times.
+
+```{r}
+obs_cens <- epidist::observe_process(sierra_leone_ebola_data)
+```
+
+Specially, the columns added by `observe_process` are:
+
+* `ptime_daily`: As the `ptime` was already daily then this is superfluous
+* `ptime_lwr`: The lower bound on the primary observation time i.e. the floor of `ptime_daily`
+* `ptime_upr`: The upper bound on the primary observation time i.e. `ptime_lwr + 1`
+* `stime_daily`: As the `ptime` was already daily then this is superfluous
+* `stime_lwr`: The lower bound on the secondary observation time i.e. the floor of `stime_daily`
+* `stime_upr`: The upper bound on the secondary observation time i.e. `stime_lwr + 1`
+* `delay_daily`: Is given by `stime_daily - ptime_daily`
+* `delay_lwr`: Is given by `stime_lwr - ptime_upr` i.e. the minimum possible delay
+* `delay_upr`: Is given by `stime_upr - ptime_lwr` i.e. the maximum possible delay
+* `obs_at`: The maximum value of `stime`
+
+## Model fitting
+
+Now we are ready to fit lognormal delay distributions to the data!
+First, we start with a simple model that assumes the same delay distribution for all cases.
+
+```{r}
+subsample <- 0.2
+```
+
+For the time being, we filter the data to only complete cases (i.e. rows of the data which have no missing values^[It should be a relatively simple extension to allow for missing data in the model.]).
+As well, to speed up computation, we take a random `r 100 * subsample`% subsample of the complete data.
+
+```{r}
+set.seed(1)
+n <- nrow(obs_cens)
+obs_cens <- obs_cens[complete.cases(obs_cens)]
+n_complete <- nrow(obs_cens)
```

Comment:
I really don't think we want a mix of dplyr and data.table in a vignette especially

Reply:
Agree as before. Resolving to have one open.